### PR TITLE
Implement full NK1 segment

### PR DESCRIPTION
**Pull request by doconnor-clintel**
https://hl7-definition.caristix.com/v2/HL7v2.3.1/Segments/NK1

### CODE REVIEW

**Comment by Stratus3D** [comment]
Hi @doconnor-clintel , thanks for the PR! I don't really maintain this project anymore. @ruby-hl7/maintainers can someone take a look?

File: lib/segments/nk1.rb [hunk 1]
```diff
@@ -1,13 +1,47 @@
...
+  end
+  add_field :end_date do |value|
+    convert_to_ts(value)
+  end
+  add_field :job_title
+  add_field :job_code
+  add_field :employee_number
   add_field :organization_name, :idx => 13
+  add_field :marital_status
+  add_field :sex # TODO: raise HL7::InvalidDataError.new( "bad administrative sex value (not F|M|O|U|A|N|C)" ) ?
```
Comment (LucasMontorio) [inline lib/segments/nk1.rb:22]: **todo:** Why not doing it in the PR directly? :)

**Review by LucasMontorio — CHANGES_REQUESTED** [review]
Hi! 
I know that this PR is quite old but I'm here to help with it :)

I have 3 main comments: 
- I suggest that we support the latest NK1 segment spec as defined in Caristix [here](https://hl7-definition.caristix.com/v2/HL7v2.3.1/Segments/NK1)
- This new segment requires specific specs using the newly-introduced fields
- The `TODO` introduced here could be implemented directly




I suggest that we support the latest NK1 segment spec as defined in Caristix [here](https://hl7-definition.caristix.com/v2/HL7v2.3.1/Segments/NK1)

**Comment by LucasMontorio** [comment]
Segment implemented in this PR: https://github.com/ruby-hl7/ruby-hl7/pull/142

**Pull request closed without merging**